u_handshake_messages.go

```diff
@@ -47,3 +51,83 @@ func (m *compressedCertificateMsg) unmarshal(data []byte) bool {
 	}
 	return true
 }
+
+type utlsEncryptedExtensionsMsgExtraFields struct {
```

Comment:
Should we be exporting these fields? Is it something a using application might want to be able to view?

Reply:
I don't believe so. `customExtension` is just a fake place holder that isn't in any published draft/standard, while `applicationSettings` could be set in `Config` for client and read from `PeerApplicationSettings` in `ConnectionState`.